src/libsyntax/ast.rs

```diff
@@ -2496,11 +2496,11 @@ pub enum ItemKind {
     /// A static item (`static`).
     ///
     /// E.g., `static FOO: i32 = 42;` or `static FOO: &'static str = "bar";`.
-    Static(P<Ty>, Mutability, P<Expr>),
+    Static(P<Ty>, Mutability, Option<P<Expr>>),
```

Comment:
The option here is also not necessary given https://github.com/rust-lang/rust/pull/69194#issuecomment-586637192.
(But it can be parsed and immediately reported for diagnostics though.)

Reply:
I don't follow. If we have `StaticCommon` then we need to have the cover grammar of every place that a `static` occurs (which would be 2 places, free and in extern blocks), so that makes the `Option<_>` necessary. This should be the case, even if we reject `= expr;` in one case and accept it in another (but that's counter to the refactorings in e.g., https://github.com/rust-lang/rust/pull/68728 I've been doing so far).

Reply:
Ah, sorry, ignore this, I assumed `static`s in both contexts require an initializer, but foreign statics actually do not.

Reply:
Ah... phew. 😅